src/migration/migration.cpp

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <filesystem>
+#include <fstream>
+
+#include "migration/migration.hpp"
+
+using namespace Poco::Data::Keywords;
+
+Migration::Migration(Poco::Data::Session& session, const std::string& migrationDir)
+    : session_(session)
+    , migrationDir_(migrationDir)
+{
+    CreateVersionTableIfNeeded();
+}
+
+void Migration::MigrateToVersion(int targetVersion)
+{
+    int currentVersion = GetCurrentVersion();
+
+    if (currentVersion == targetVersion) {
+        return;
+    }
+
+    if (currentVersion < targetVersion) {
+        UpgradeDatabase(targetVersion, currentVersion);
+
+        return;
+    }
+
+    DowngradeDatabase(targetVersion, currentVersion);
+}
+
+void Migration::ApplyMigration(const std::string& migrationScript)
+{
+    std::string   path = std::filesystem::path(migrationDir_) / migrationScript;
+    std::ifstream file(path);
+
+    if (!file.is_open()) {
+        throw std::runtime_error("Failed to open migration script: " + migrationScript);
```

Comment:
Should we move AosException to this lib and use it here instead of std::exception?

Reply:
Done